### PR TITLE
Refactor spec setup to use :apply_to_host_groups

### PR DESCRIPTION
**Pull request by pirj**
See: https://github.com/rspec/rspec-core/issues/2832

### CODE REVIEW

File: spec/spec_helper.rb [hunk 1]
```diff
@@ -31,35 +31,7 @@ def hash_inspect(hash)
   end
 end
 
-RSpec.configure do |config|
```
Comment (pirj) [inline spec/spec_helper.rb:34]: Had to move this below shared contexts.

File: spec/spec_helper.rb [hunk 2]
```diff
@@ -128,4 +99,37 @@ def with_minitest_loaded
...
+    expectations.syntax = :expect
+    expectations.include_chain_clauses_in_custom_matcher_descriptions = true
+    expectations.strict_predicate_matchers = true
+  end
+
+  config.mock_with :rspec do |mocks|
+    mocks.verify_partial_doubles = true
+  end
+
+  config.include_context "with #should enabled", :uses_should
```
Comment (pirj) [inline spec/spec_helper.rb:120]: Because they are included here now.

File: spec/spec_helper.rb [hunk 3]
```diff
@@ -128,4 +99,37 @@ def with_minitest_loaded
...
+  config.include_context "with #should exclusively enabled", :uses_only_should
+  config.include_context "with warn_about_potential_false_positives set to false", :warn_about_potential_false_positives
+
+  config.shared_context_metadata_behavior = :apply_to_host_groups if config.respond_to?(:shared_context_metadata_behavior=) # RSpec 4 dropped this setting
```
Comment (pirj) [inline spec/spec_helper.rb:124]: And switched this to the new default.

File: spec/spec_helper.rb (hunk 1 above)
Comment (JonRowe) [reply spec/spec_helper.rb:34]: Rather than moving it, split it.

File: spec/spec_helper.rb (hunk 2 above)
Comment (JonRowe) [reply spec/spec_helper.rb:120]: Only these need to be below, the rest can stay up top.

File: spec/spec_helper.rb (hunk 3 above)
Comment (JonRowe) [reply spec/spec_helper.rb:124]: The rest could go back up the top

**Review by JonRowe — DISMISSED** [review]

**Pull request merged**